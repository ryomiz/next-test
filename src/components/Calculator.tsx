import { css } from '@emotion/react'

import { CalculatorButton } from './CalculatorButton'

import type { Operators } from 'src/types'

import { useCalcurator } from '@/hooks/useCalculator'

const arr = [...Array(9)].map((_, index: number) => index + 1)
const symbols: Operators[] = ['+', '-', '*', '/']

export const Calculator: React.VFC = () => {
  const {
    firstVal,
    secondVal,
    result,
    operator,
    setValue,
    handleOperator,
    calculate,
    clear,
  } = useCalcurator()

  return (
    <div>
      <div css={display}>
        {result ? result : `${firstVal} ${operator} ${secondVal}`}
      </div>
      <div css={flex}>
        <div css={grid}>
          {arr.map((number) => (
            <CalculatorButton key={number} onClick={() => setValue(number)}>
              {number}
            </CalculatorButton>
          ))}
          <CalculatorButton onClick={clear}>C</CalculatorButton>
          <CalculatorButton onClick={() => setValue(0)}>0</CalculatorButton>
          <CalculatorButton onClick={calculate}>=</CalculatorButton>
        </div>
        <div css={column}>
          {symbols.map((sym) => (
            <CalculatorButton key={sym} onClick={() => handleOperator(sym)}>
              {sym}
            </CalculatorButton>
          ))}
        </div>
      </div>
    </div>
  )
}

const display = css`
  padding: 0 0.2em;
  border: 1px solid #ccc;
  margin-bottom: 20px;
  font-size: 5rem;
  text-align: right;
`

const flex = css`
  display: flex;
  gap: 20px;
`

const grid = css`
  display: grid;
  height: 200px;
  flex: 3;
  justify-content: end;
  gap: 20px;
  grid-template-columns: repeat(3, 1fr);
`

const column = css`
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 20px;
`
