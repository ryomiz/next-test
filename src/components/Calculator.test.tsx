import { cleanup, render, screen } from '@testing-library/react'
import { Calculator } from './Calculator'

describe('Calculator Test', () => {
  afterEach(() => {
    cleanup()
  })

  // スナップショットテスト
  test('render', () => {
    const { asFragment } = render(<Calculator />)
    expect(asFragment()).toMatchSnapshot()
  })
})
