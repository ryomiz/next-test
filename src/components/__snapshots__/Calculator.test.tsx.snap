// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-1a5o32j-display"
    >
      0  0
    </div>
    <div
      class="css-14cdcmy-flex"
    >
      <div
        class="css-183dgzn-grid"
      >
        <button
          class="css-qmwc4l-button"
        >
          1
        </button>
        <button
          class="css-qmwc4l-button"
        >
          2
        </button>
        <button
          class="css-qmwc4l-button"
        >
          3
        </button>
        <button
          class="css-qmwc4l-button"
        >
          4
        </button>
        <button
          class="css-qmwc4l-button"
        >
          5
        </button>
        <button
          class="css-qmwc4l-button"
        >
          6
        </button>
        <button
          class="css-qmwc4l-button"
        >
          7
        </button>
        <button
          class="css-qmwc4l-button"
        >
          8
        </button>
        <button
          class="css-qmwc4l-button"
        >
          9
        </button>
        <button
          class="css-qmwc4l-button"
        >
          C
        </button>
        <button
          class="css-qmwc4l-button"
        >
          0
        </button>
        <button
          class="css-qmwc4l-button"
        >
          =
        </button>
      </div>
      <div
        class="css-1yi6v5b-column"
      >
        <button
          class="css-qmwc4l-button"
        >
          +
        </button>
        <button
          class="css-qmwc4l-button"
        >
          -
        </button>
        <button
          class="css-qmwc4l-button"
        >
          *
        </button>
        <button
          class="css-qmwc4l-button"
        >
          /
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
